### ISSUE TITLE
/wb fill very slow on 1.14.4

### ISSUE THREAD

**Issue by willc5** [Brettflan/WorldBorder#153]
Only 1% of the world generated overnight at default fill command. How can I make it faster? No point in pregenerating if it's gonna take a month

**Comment by willc5** [comment]
And can I change the fill frequency while it's running or paused

**Comment by FrostDelta123** [comment]
Same. Paper-spigot 1.14.4. 30x30k map. 40% by 7 days
20GB RAM doesnt help. screen http://prntscr.com/owiyzl

**Comment by BoomEaro** [comment]
I confirm, the world is being generated too slowly.
The interesting thing is that if you stop generating the world and start it again, then the plugin will quickly generate chunks for some time. However, then it will again generate slowly. Moreover, at these moments, the load on the CPU is very high.

**Comment by Tsoccerguy3** [comment]
do /timings report , to see if there is a mem leak or a bad plugin

**Comment by mxs42** [comment]
@BoomEaro it doesn't generate quickly on start, it just doesn't regenerate what already was generated. It skips sectors and then starts generating normally, I think this is can not be fixed because world generation is slow in general in 1.14.

**Comment by dylantknguyen** [comment]
Seems to be a memory leak in Spigot 1.14.4. After reverting back to [this release of Paper-1.14.4](https://papermc.io/ci/job/Paper-1.14/83/), the problem seemed to be gone. I plan to let it generate for the next few days before switching back to my normal jar.

**Comment by ShiftingKill** [comment]
Can anyone confirm if leaving a player AFK somewhere helps this issue? Just noticing now that the hundreds/sec and thousands/sec are more frequent with an AFK admin on. Noticeable enough to not want to restart every 10 minutes. @FrostDelta123 I also am around that same setup but mine is 20k circular radius and its a 1/3 of the way done in 2 days.

**Comment by Puremin0rez** [comment]
I honestly think there's either a memory leak in WorldBorder or a memory leak in the chunk loading system somewhere.

I'm generating a 50k x 50k (25k each direction) border and while I do realize this will take days to complete, I've had to resort to setting up a cron job to restarting the server hourly to recover the memory that is seemingly being permanently eaten. The chunk generation starts off quick enough (100-120~ chunks) and then drops quite drastically as the memory starts to get lower (10-30~ chunks) and will complain about freeing up memory quite often. This is with 20GB allocated.

**Comment by a3s** [comment]
For future readers: use [chunkmaster](https://github.com/Trivernis/spigot-chunkmaster).
WorldBorder is buggy and I don't see how it can be useful nowadays.
Chunkmaster works fast and without any issues, world border is a vanilla future now.

tldr: do not use this plugin
